pkg/controller/certificates/issuing/internal/keystore.go

```diff
@@ -59,17 +60,37 @@ func encodePKCS12Keystore(password string, rawKey []byte, certPem []byte, caPem
 	if len(certs) > 1 {
 		cas = append(certs[1:], cas...)
 	}
-	return pkcs12.LegacyRC2.Encode(key, certs[0], cas, password)
+
+	switch algorithm {
+	case cmapi.AESPKCS12Algorithm:
+		return pkcs12.Modern2023.Encode(key, certs[0], cas, password)
```

Comment:
suggestion: If we're using a variable named "Modern2023" there's no indication there that it'll always use AES. They could use ChaCha and I'd still call it modern.

I don't think the underlying library is likely to change the algorithm used for `Modern2023` but it's a risk to call it AES in our API I think.

Reply:
@SgtCoDFish, I completely share your sentiments regarding the mapping of the encoders from `software.sslmate.com/src/go-pkcs12`. It was quite a challenge to create a user-friendly API interface that abstracts these details. 

My concern is that the average cert-manager user may not be familiar with the specifics of this Go package's implementation, and we should ensure that the API remains intuitive and accessible.

Reply:
We also had a few other discussions. I don't think it's worth abstracting something that we can't abstract. Let's map it 1:1 to at least provide all the capabilities.

Reply:
Yeah I think what you're saying makes sense. If the names match the underlying lib I think that's fine - we can always make it more resilient down the road if we want 😁

Reply:
Do you have an idea for a regression test regarding the encoded PKCS#12? Perhaps something similar to `openssl pkcs12 -info` could help to check the algorithms used.

Reply:
I'd definitely rather avoid adding a dependency on openssl for our tests - but I don't have a better suggestion than that, since it looks like go-pkcs12 doesn't expose this info.

Maybe we could copy paste a bit of their parsing code - but for now, I'm not too bothered about it. I'd be a nice thing to add later but I don't think it's a hard requirement really... at the end of the day we discourage these passwords!

Reply:
I tried to use `encoding/asn1` to retrieve the information, but I stopped before I coded my own version of `software.sslmate.com/src/go-pkcs12` 🙈